lib/apis/moray.js

```diff
@@ -492,7 +492,33 @@ Moray.prototype._createSearch = function (params, cb) {
     });
 };
 
+/**
+ * List all VMs for a given server from the VM bucket.
+ * @param  {string}     uuid uuid of the server whos vms should be fetched.
+ * @param  {Function}   cb  function of the form function(err, vms) {...}
+ */
+Moray.prototype.listVmsForServer = function listVmsForServer(uuid, cb) {
+    var self = this;
+    var vms = {};
+    var vm;
+    var filter = sprintf(PARAM_FILTER, 'server_uuid', uuid);
```

Comment:
@joshwilsdon I updated the PR to filter out destroyed VMs. That said, I'm punting on upping the limit parameter or paginating the query, because of the work we'll be doing to remove the PUT /vms route all together. Also it seems unlikely that we'll see >1000 active VMs on a CN before we remove the PUT /vms functionality altogether.